components/places/android/src/main/java/mozilla/appservices/places/Bookmarks.kt

```diff
@@ -119,6 +119,19 @@ interface ReadableBookmarksConnection : InterruptibleConnection {
      * has its `interrupt()` method called on another thread.
      */
     fun getRecentBookmarks(limit: Int): List<BookmarkItem>
+
+    /**
+     * Counts the number of bookmark items in the bookmark trees under the specified GUIDs.
+
+     * @param guids The guids of folders to query.
+     * @return Count of all bookmark items (ie, not folders or separators) in all specified folders recursively.
+     * Empty folders, non-existing GUIDs and non-existing items will return zero.
+     * The result is implementation dependant if the trees overlap.
```

Comment:
Yeah - my first version took just one arg, but then looking at the code in a-c it was clear that it would always need to sum the values from Menu, Toolbar and Unfiled, then use a separate count for Mobile - so I figured I'd optimize for the actual use-case. FWIW, https://github.com/mhammond/firefox-android/tree/count-bookmarks-in-trees is that branch.